### PR TITLE
fix(GraphQL Node): Throw error if GraphQL variables are not objects or strings

### PR DESCRIPTION
**Pull request by dana-gill**
## Summary

This PR throws an error if the variables set in the GraphQL nodes are neither objects nor strings. This also performs some cleanups to the GraphQL node.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/NODE-1602/graphql-node-unsuccessful-evaluation-of-variables-expression-silently

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/nodes-base/nodes/GraphQL/GraphQL.node.ts [hunk 1]
```diff
@@ -418,40 +418,45 @@ export class GraphQL implements INodeType {
...
-									'Using variables failed:\n' +
-										(jsonBody.variables as string) +
-										'\n\nWith error message:\n' +
-										(error as string),
-									{ itemIndex },
-								);
-							}
-						}
-						if (jsonBody.operationName === '') {
-							jsonBody.operationName = null;
```
Comment (elsmr) [inline packages/nodes-base/nodes/GraphQL/GraphQL.node.ts:448]: I think this behavior was important. With the Linear GraphQL API for example this throws an error now.
You can use the node from the ticket description to reproduce it.

**Review by elsmr — CHANGES_REQUESTED** [review]
Good work on the test and cleaning up the code 🙌 I did find a couple blocking bugs

**Comment by dana-gill** [comment]
PTAL! Specifically the last two commits. If everything looks fine then I'll squash them :)

**Review by elsmr — APPROVED** [review]
LGTM!

**Review by Joffcom — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.71.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.71.0)

**Pull request merged**